src/Native/System.Net.Security.Native/pal_gssapi.cpp

```diff
@@ -144,22 +155,42 @@ extern "C" uint32_t NetSecurityNative_InitSecContext(uint32_t* minorStatus,
     assert(contextHandle != nullptr);
     assert(outBuffer != nullptr);
     assert(retFlags != nullptr);
+    assert(isNtlmUsed != nullptr);
     assert(inputBytes != nullptr || inputLength == 0);
 
 // Note: claimantCredHandle can be null
 
 #if HAVE_GSS_SPNEGO_MECHANISM
-    gss_OID desiredMech = isNtlm ? GSS_NTLM_MECHANISM : GSS_SPNEGO_MECHANISM;
+    gss_OID desiredMech;
+    if (isNtlm)
+    {
+        desiredMech = GSS_NTLM_MECHANISM;
+    }
+    else
+    {
+        desiredMech = GSS_SPNEGO_MECHANISM;
+    }
+
+    gss_OID krbMech = GSS_KRB5_MECHANISM;
 #else
-    assert(!isNtlm && "NTLM is not supported by MIT libgssapi_krb5");
-    (void)isNtlm; // unused
-
-    gss_OID_desc gss_mech_spnego_OID_desc = {.length = 6, .elements = static_cast<void*>(gss_mech_value)};
-    gss_OID desiredMech = &gss_mech_spnego_OID_desc;
+    gss_OID_desc gss_mech_OID_desc;
+    if (isNtlm)
+    {
+        gss_mech_OID_desc = { .length = 10,.elements = static_cast<void*>(gss_ntlm_mech_value) };
```

Comment:
Can we get these magic numbers converted to constants or compiler-generated values?